Objects/object.c

```diff
@@ -663,6 +663,32 @@ PyObject_Bytes(PyObject *v)
     return PyBytes_FromObject(v);
 }
 
+
+/*
+def _PyObject_FunctionStr(f):
+    try:
+        return f.__qualname__ + "()"
+    except Exception:
+        return type(f).__name__ + " object"
+*/
+PyObject *
+_PyObject_FunctionStr(PyObject *f)
+{
```

Comment:
Please add: assert(!PyErr_Occurred()); since this function can replace the current exception.